backend/schema/schema.go

```diff
@@ -90,7 +96,8 @@ func (s *Schema) ResolveRefToType(ref *Ref, out Decl) error {
 			}
 		}
 	}
-	return fmt.Errorf("could not resolve reference %v", ref)
+
+	return errors.Join(ErrNotFound, fmt.Errorf("could not resolve reference %v", ref))
```

Comment:
Ah also please use `fmt.Errorf` and `%w`, as `errors.Join()` formats terribly

```suggestion
	return fmt.Errorf("could not resolve reference %v: %w", ref, ErrNotFound)
```

Reply:
oh forgot about that one. makes sense.